### PR TITLE
Mon pdb reconcile post OCS upgrade

### PR DESCRIPTION
**Pull request by Shrivaibavi**
Automated BZ https://bugzilla.redhat.com/show_bug.cgi?id=1888713

### CODE REVIEW

File: ocs_ci/ocs/constants.py [hunk 1]
```diff
@@ -119,6 +119,8 @@
 VOLUMESNAPSHOTCLASS = "VolumeSnapshotClass"
 HPA = "horizontalpodautoscaler"
 VOLUMESNAPSHOTCONTENT = "VolumeSnapshotContent"
+POD_DISRUPTION_BUDGET = "pdb"
```
Comment (prsurve) [inline ocs_ci/ocs/constants.py:122]: try using the full name for the kind

**Review by Avilir — APPROVED** [review]

File: ocs_ci/ocs/constants.py (hunk 1 above)
Comment (Shrivaibavi) [reply ocs_ci/ocs/constants.py]: Done

**Review by hnallurv — DISMISSED** [review]

File: tests/ecosystem/upgrade/test_upgrade.py [hunk 2]
```diff
@@ -52,3 +54,25 @@ def test_upgrade():
     """
 
     run_ocs_upgrade()
+
+
+@ocs_upgrade
```
Comment (ebenahar) [inline tests/ecosystem/upgrade/test_upgrade.py:59]: this marker means that this test is an upgrade test. I think it should not be in use here. 
@AviadP @petr-balogh could you please review?

Comment (Shrivaibavi) [reply tests/ecosystem/upgrade/test_upgrade.py]: changed to post_ocs_upgrade

File: ocs_ci/helpers/helpers.py [hunk 3]
```diff
@@ -2790,3 +2790,24 @@ def fetch_used_size(cbp_name, exp_val=None):
             f"matching. Retrying"
         )
     return used_in_gb
+
+
+def get_mon_pdb():
```
Comment (ebenahar) [inline ocs_ci/helpers/helpers.py:2795]: sorry for not raising it earlier, but how about making this function generic, to get any type of pod, such as OSD PDB, MDS, etc.

What do you think?

Comment (Shrivaibavi) [reply ocs_ci/helpers/helpers.py:2795]: I see this PR already address this, https://github.com/red-hat-storage/ocs-ci/pull/1260/files Do you still need this function to be made generic ? Please correct me if my understanding is wrong

Comment (ebenahar) [reply ocs_ci/helpers/helpers.py:2795]: so maybe we can merge these 2 together into a single function that gets the pod type and returns the PDB?

Comment (Shrivaibavi) [reply ocs_ci/helpers/helpers.py:2795]: ACK

Comment (ebenahar) [reply ocs_ci/helpers/helpers.py:2795]: this could be done separately, please open an issue to track the needed change

**Review by ebenahar — DISMISSED** [review]

File: ocs_ci/helpers/helpers.py (hunk 3 above)
Comment (Shrivaibavi) [reply ocs_ci/helpers/helpers.py:2795]: https://github.com/red-hat-storage/ocs-ci/issues/3634

File: tests/ecosystem/upgrade/test_upgrade.py [hunk 4]
```diff
@@ -52,3 +54,25 @@ def test_upgrade():
...
+    # Check for mon count
+    mons_after_upgrade = ceph_obj.get_mons_from_cluster()
+    log.info(f"Mons after upgrade {mons_after_upgrade}")
+
+    disruptions_allowed, min_available_mon = get_mon_pdb()
+    log.info(f"Number of Mons Disruptions_allowed {disruptions_allowed}")
+    log.info(f"Minimum_available mon count {min_available_mon}")
+
+    assert disruptions_allowed == 1, "Mon Disruptions_allowed count not matching"
+    assert min_available_mon == 2, "Minimum available mon count is not matching"
```
Comment (PrasadDesala) [inline tests/ecosystem/upgrade/test_upgrade.py:78]: the mon pdb values got changed from 4.5 on wards, may be add a comment about it just for info purpose

File: tests/ecosystem/upgrade/test_upgrade.py [hunk 5]
```diff
@@ -52,3 +54,25 @@ def test_upgrade():
     """
 
     run_ocs_upgrade()
+
+
+@post_ocs_upgrade
+@pytest.mark.polarion_id("OCS-2449")
+def test_check_mon_pdb_post_upgrade():
+    """
+    Function to check disruptions_allowed and minimum
```
Comment (PrasadDesala) [inline tests/ecosystem/upgrade/test_upgrade.py:63]: this is not a function but part of test we are checking. Please correct it

**Review by PrasadDesala — COMMENTED** [review]
LGTM. Just  minor comments and also rebase to master and resolve conflicts.

File: tests/ecosystem/upgrade/test_upgrade.py (hunk 4 above)
Comment (Shrivaibavi) [reply tests/ecosystem/upgrade/test_upgrade.py:90]: Done

**Review by OdedViner — APPROVED** [review]
LGTM

**Review by ebenahar — APPROVED** [review]

**Review by shreekarSS — APPROVED** [review]

**Pull request merged**